### DISCUSSION TITLE
Get m5stack Core2 ENV III sensor values

### DISCUSSION THREAD

**Discussion by Slyke** [arendst/Tasmota#14359] [Q&A] [answered]
Hello, something happened to my last thread of this question.

According to this discussion: https://github.com/arendst/Tasmota/discussions/13833 Tasmota should be getting the temperature values from the SHT3x.

I flashed Tasmota onto a M5stack Core2 and connected an I2C ENV III sensor to it, but Tasmota is unable to get sensor readings from it.

I've tested the sensor it self with the example code provided by the manufacturer, and it does correctly get values. I've also run the I2C scan in Tasmota and can see that it can see both sensors inside the ENV III device:
```
22:25:47.076 CMD: I2Cscan
22:25:47.096 RSL: RESULT = {"I2CScan":"Device(s) found at 0x44 0x70"}
```

The temperature and humidity (SHT3x) sensor is 0x44 and the pressure (QMP6988) sensor is 0x70, according to this:
https://shop.m5stack.com/products/env-iii-unit-with-temperature-humidity-air-pressure-sensor-sht30-qmp6988

On the WUI on Tasmota I can see a bunch of sensor details, but none of them are the readings from the ENV III device:
![image](https://user-images.githubusercontent.com/6442613/148741237-11d1a206-9a62-4b05-9267-603295171c1f.png)

Tasmota normally automatically detects the type of sensor and displays it on the main page, but it doesn't seem to be in this case.

**Comment by sfromis** [comment, marked as answer]
Tasmota does not support the QMP6988 sensor until someone creates a driver, native or in Berry.

**Reply by Slyke** [reply]
Yep, but it should be able to get the data from the SHT3x though, right?

**Reply by sfromis** [reply]
Yeah, it is two different sensors. SHT3X support depends on you using a binary including this feature. You can check if driver number 15 is included in the output of [I2CDriver](https://tasmota.github.io/docs/Commands/#i2cdriver).

**Reply by Slyke** [reply]
Running `I2CDriver15 1` produces:
```11:49:38.039 CMD: I2CDriver15 1
11:49:38.044 RSL: RESULT = {"I2CDriver":"3,4,5,7,47,58"}
```

But, I searched for `USE_SHT3X` after following the documentation you linked, and found `//  #define USE_SHT3X`: https://github.com/arendst/Tasmota/blob/development/tasmota/my_user_config.h#L588

Uncommented it out, recompiled and flashed and now it's showing up!

As for the `QMP6988` sensor, looks like it isn't supported. I'll see if I can write the driver for it, since there is already the C source for the driver from the manufacturer ( https://github.com/m5stack/UNIT_ENV/blob/master/src/QMP6988.cpp ) , but not sure if my C/C++ is up to snuff.

Thank you!
